liquid-rust-typeck/src/checker.rs

```diff
@@ -302,6 +305,33 @@ impl<'a, 'tcx> Checker<'a, 'tcx> {
         Ok(())
     }
 
+    fn check_assert(
+        &mut self,
+        env: &mut TypeEnv<'tcx>,
+        cursor: &mut Cursor,
+        cond: &Operand,
+        expected: &bool,
+        target: BasicBlock,
+    ) -> Result<(), ErrorReported> {
+        let cond_ty = self.check_operand(env, cond);
+
+        let pred = match cond_ty.kind() {
+            TyKind::Refine(BaseTy::Bool, e) => {
+                e.clone()
+            }
+            _ => unreachable!("unexpected cond_ty {:?}", cond_ty),
+        };
+
+        let assert = ExprKind::BinaryOp(BinOp::Eq, pred, ExprKind::Constant(liquid_rust_fixpoint::Constant::Bool(*expected)).intern()).intern();
+
+        // Uncomment the below line to allow pre-catching of possible divide-by-zero, underflow, and overflow
+        // WARNING: rust is very eager about inserting under/overflow checks, so be warned that uncommenting this will likely break everything
```

Comment:
More than a `warning` I would add a `TODO` to make this user-configurable. We have some support for config flags here https://github.com/liquid-rust/liquid-rust/blob/main/liquid-rust-common/src/config.rs so we could make a new config to choose between both. It doesn't have to go in this PR though.